go-controller/pkg/ovn/policy.go

```diff
@@ -217,23 +220,31 @@ func deleteACLPortGroup(portGroupName, direction, priority, match, action string
 	return nil
 }
 
-func addToPortGroup(portGroup string, portInfo *lpInfo) error {
-	_, stderr, err := util.RunOVNNbctl("--if-exists", "remove",
-		"port_group", portGroup, "ports", portInfo.uuid, "--",
-		"add", "port_group", portGroup, "ports", portInfo.uuid)
-	if err != nil {
-		return fmt.Errorf("failed to add logicalPort %s to portGroup %s "+
-			"stderr: %q (%v)", portInfo.name, portGroup, stderr, err)
+func addToPortGroup(ovnNBClient goovn.Client, portGroupName string, portInfo *lpInfo) error {
+	cmd, err := ovnNBClient.PortGroupAddPort(portGroupName, portInfo.uuid)
+	if err == nil {
+		if err = ovnNBClient.Execute(cmd); err != nil {
+			return fmt.Errorf("execute error adding port %s to port group %s (%v)", portInfo.name, portGroupName, err)
+		}
+	} else if (err != goovn.ErrorExist) && (err != goovn.ErrorNotFound) {
```

Comment:
Shouldnt the check for err != goovn.ErrorExist be part of the else if of the inner if where the cmd is executed.

Reply:
The checking is actually done in the PortGroupAddPort() call, so this is the right place for it.

Reply:
Yes..you are right.